app/controllers/api/configuration_script_payloads_controller.rb

```diff
@@ -1,5 +1,18 @@
 module Api
   class ConfigurationScriptPayloadsController < BaseController
     include Subcollections::Authentications
+
+    def edit_resource(type, id, data)
+      resource = resource_search(id, type)
+
+      allowed_params  = %w[description credentials]
```

Comment:
Do we also verify that the user has RBAC permission to the credentials they are trying to set?

Reply:
That mapping to authentication records is done separately via the `configuration_script_payload/:id/authentications` subcollection, this is about credential references in the ASL payload are mapped to values in those authentications so I don't think RBAC would come into play here here, that would have already taken place when they mapped authentication records.

Example credentials payload:
`{"credentials":{"api_user":"$.manageiq_api.userid", "api_password":"$.manageiq_api.password"}}`

Reply:
Now that we're automatically updating the authentications join table we are running RBAC on the requested credentials below.